map_unix.go

```diff
@@ -8,23 +8,21 @@ import (
 	"golang.org/x/sys/unix"
 )
 
-type slice struct {
-	addr unsafe.Pointer
-	len  int
-	cap  int
-}
-
 func mmap(size uint) (unsafe.Pointer, error) {
 	b, err := unix.Mmap(-1, 0, int(size), unix.PROT_READ|unix.PROT_WRITE, unix.MAP_PRIVATE|unix.MAP_ANONYMOUS)
 	if err != nil {
 		return nil, err
 	}
-	sl := (*slice)(unsafe.Pointer(&b))
-	return sl.addr, nil
+	return unsafe.Pointer(&b[0]), nil
 }
 
 func munmap(p unsafe.Pointer) error {
 	size := int(((*header)(p)).size + szheader)
-	b := *(*[]byte)(unsafe.Pointer(&slice{p, size, size}))
+	var sl = struct {
+		addr unsafe.Pointer
+		len  int
+		cap  int
+	}{p, size, size}
+	b := *(*[]byte)(unsafe.Pointer(&sl))
```

Comment:
This should be:

    var b []byte
    sl := (*reflect.SliceHeader)(unsafe.Pointer(&b))
    sl.Data = uintptr(p)
    sl.Len = size
    sl.Cap = size

Reply:
Thanks for looking at this. I can send a follow-up patch to x/sys, stdlib, etc... that addresses these situations.